src/Features/Core/Portable/SemanticSearch/Tools/ReferencingSyntaxFinder.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if NET6_0_OR_GREATER
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Notification;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Threading;
+
+namespace Microsoft.CodeAnalysis.SemanticSearch;
+
+internal sealed class ReferencingSyntaxFinder(Solution solution, OptionsProvider<ClassificationOptions> classificationOptions, CancellationToken cancellationToken)
+{
+    public IEnumerable<SyntaxNode> FindSyntaxNodes(ISymbol symbol)
+        => FindSyntaxNodesAsync(symbol).ToBlockingEnumerable(cancellationToken);
+
+    public async IAsyncEnumerable<SyntaxNode> FindSyntaxNodesAsync(ISymbol symbol)
```

Comment:
i think i prefer FindReferencingSyntaxNodes and FindReferencingSyntaxNodesAsync.  how do you feel?

Reply:
Seems redundant since the containing type is already called ReferencingSyntaxFinder. Maybe just Find :)

Reply:
oh true.  i was thinking this was the name a *user* saw.  

1. i'm fine with this being called 'Find'
2. i like the name you chose (FindReferencingSyntaxNodes) for what users see.

#resolved.